src/components/Profile/Profile.js

```diff
@@ -50,9 +54,50 @@ class UserProfile extends Component {
       }));
     }
   }
+
+  handleSubmit = e => {
+    e.preventDefault();
+    if ( this.state.newpassword!=="" && this.state.confnewpassword==="" ) {
+      this.setState({
+        alertMessage: "Please confirm your password !!!"
+      })
+      return;
+    }
+    if ( this.state.newpassword==="" && this.state.confnewpassword!=="" ) {
+      this.setState({
+        alertMessage: "Please enter both password fields !!!"
+      })
+      return;
+    }
+    if ( this.state.newpassword!==this.state.confnewpassword ) {
+      this.setState({
+        alertMessage: "Password do not match !!!"
+      })
+      return;
+    }
+    let body={};
+    if ( this.state.name!=="" && this.state.name!==this.state.initialName ) {
+      body.name=this.state.name;
+    }
+    if ( this.state.email!=="" && this.state.email!==this.state.initialEmail ) {
+      body.email=this.state.email;
+    }
+    if ( this.state.newpassword!=="" ) {
+      body.password=this.state.newpassword;
+    }
+    this.setState({
+      alertMessage: null
+    })
+    const update={
```

Comment:
Set update object and Make the call only if body is not an empty object. Put that check.